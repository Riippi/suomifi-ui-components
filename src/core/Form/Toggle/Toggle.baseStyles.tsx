import { css } from 'styled-components';
import { suomifiTheme } from '../../theme';
import { focus } from '../../theme/utils';
import { ToggleProps } from './Toggle';
import { element, fonts } from '../../theme/reset';

const svgPrefix = 'icon-toggle_svg__';

export const baseStyles = ({ theme = suomifiTheme }: ToggleProps) => css`
  ${element(theme)}
  ${fonts(theme).body}
  background-color: ${theme.colors.whiteBase};
  & > .fi-toggle_label {
    cursor: pointer;
  }
  & > .fi-toggle_input {
    ${element(theme)}
    ${fonts(theme).body}
    width: 0;
    height: 0;
    opacity: 0;
    z-index: -9999;
    background-color: ${theme.colors.whiteBase};
    &:focus {
      outline: 0;
      & + .fi-toggle_label {
        ${focus({ theme, noPseudo: true })}
      }
    }
    &:focus:not(:focus-visible) {
      outline: 0;
      &:after {
        content: none;
      }
    }
  }
  & .fi-toggle_icon {
    width: 40px;
    height: 24px;
    margin-right: ${theme.spacing.s};
    vertical-align: bottom;
    overflow: visible;
    transform: translateY(-0.1em);
  
    & * {
      cursor: pointer;
    }
  
    & .${svgPrefix}fi-toggle-icon-knob {
      transform: translateX(0%);
    }
    & .${svgPrefix}fi-toggle-icon-slide {
      transform: translateY(1px);
    }
    &.fi-toggle_icon--checked {
      .${svgPrefix}fi-toggle-icon-knob {
        transform: translateX(50%);
      }
      .${svgPrefix}fi-toggle-icon-slide {
        fill: ${theme.colors.successSecondary};
      }
      .${svgPrefix}fi-toggle-icon-circle {
        fill: ${theme.colors.successBase};
      }
    }
  }
`;
