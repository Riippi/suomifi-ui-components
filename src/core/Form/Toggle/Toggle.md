```js
import { Toggle } from 'suomifi-ui-components';
<>
  <Toggle onClick={({ toggleState }) => console.log(toggleState)}>
    Toggle button
  </Toggle>
  <Toggle.withInput
    onClick={({ toggleState }) => console.log(toggleState)}
  >
    Toggle input
  </Toggle.withInput>
</>;
```
