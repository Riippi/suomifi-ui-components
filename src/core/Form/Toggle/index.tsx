export { ToggleProps } from './Toggle/Toggle';
export { ToggleInput } from './ToggleInput/ToggleInput';
export { ToggleButton } from './ToggleButton/ToggleButton';
