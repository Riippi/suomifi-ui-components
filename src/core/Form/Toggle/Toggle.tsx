import React, { Component } from 'react';
import { default as styled } from 'styled-components';
import classnames from 'classnames';
import { withDefaultTheme } from '../../theme/utils';
import { ThemeComponent } from '../../theme';
import { baseStyles } from './Toggle.baseStyles';
import {
  Toggle as CompToggle,
  ToggleProps as CompToggleProps,
  ToggleInput,
  ToggleInputProps as CompToggleInputProps,
} from '../../../components/Form/Toggle';
import { Icon } from '../../Icon/Icon';

export interface ToggleProps extends CompToggleProps, ThemeComponent {}
export interface ToggleInputProps
  extends CompToggleInputProps,
    ThemeComponent {}

const iconBaseClassName = 'fi-toggle_icon';
const iconDisabledClassName = `${iconBaseClassName}--disabled`;
const iconCheckedClassName = `${iconBaseClassName}--checked`;

const StyledToggle = styled(({ theme, ...passProps }: ToggleProps) => (
  <CompToggle {...passProps} />
))`
  ${props => baseStyles(props)}
`;

/**
 * <i class="semantics" />
 * Use for toggling form selection or application state
 */
export class Toggle extends Component<ToggleProps> {
  state = { toggleStatus: !!this.props.checked };

  handleToggle = () => {
    const { onClick } = this.props;
    const { toggleStatus } = this.state;
    this.setState({ toggleStatus: !toggleStatus });
    if (!!onClick) {
      onClick({ toggleState: !toggleStatus });
    }
  };

  render() {
    const {
      children,
      disabled = false,
      checked: dissMissChecked,
      onClick,
      ...passProps
    } = withDefaultTheme(this.props);
    const { toggleStatus } = this.state;

    return (
      <StyledToggle {...passProps} onClick={this.handleToggle}>
        <Icon
          icon="toggle"
          className={classnames(iconBaseClassName, {
            [iconDisabledClassName]: !!disabled,
            [iconCheckedClassName]: !!toggleStatus,
          })}
          mousePointer={true}
        />
        {children}
      </StyledToggle>
    );
  }
}

export { ToggleInput };
