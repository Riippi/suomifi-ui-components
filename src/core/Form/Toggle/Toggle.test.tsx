import React from 'react';
import { render } from 'react-testing-library';
import { axeTest } from '../../../utils/test/axe';

import { Toggle } from './Toggle';

const doNothing = () => ({});

const TestToggle = (
  <Toggle onClick={doNothing} data-testid="toggle" id="test-toggle">
    Test
  </Toggle>
);

test('calling render with the same component on the same container does not remount', () => {
  const buttonRendered = render(TestToggle);
  const { getByTestId, container, rerender } = buttonRendered;
  expect(container.firstChild).toMatchSnapshot();
  expect(getByTestId('toggle').textContent).toBe('Test');

  // re-render the same component with different props
  rerender(
    <Toggle onClick={doNothing} data-testid="elggot">
      Test two
    </Toggle>,
  );
  expect(getByTestId('elggot').textContent).toBe('Test two');
});

test('should not have basic accessibility issues', axeTest(TestToggle));
