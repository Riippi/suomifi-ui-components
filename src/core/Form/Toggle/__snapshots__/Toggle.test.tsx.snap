// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Toggle variant default:  calling render with the same component on the same container does not remount 1`] = `
.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  text-transform: none;
  -webkit-appearance: button;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
  cursor: pointer;
}

.c0:-moz-focusring {
  outline: 1px dotted ButtonText;
}

.c0::-moz-focus-inner {
  border-style: none;
  padding: 0;
}

.c0::-webkit-file-upload-button {
  -webkit-appearance: button;
  font: inherit;
}

.c0::-webkit-inner-spin-button {
  height: auto;
}

.c0::-webkit-outer-spin-button {
  height: auto;
}

.c3 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c1.fi-toggle--disabled {
  cursor: not-allowed;
}

.c4 {
  display: inline-block;
  vertical-align: baseline;
}

.c5 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  color: hsl(0,0%,16%);
}

.c5.fi-text--bold {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 600;
}

.c5.fi-text--lead {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 22px;
  line-height: 1.5;
  font-weight: 400;
}

.c5.fi-text--small-screen {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
}

.c5.fi-text--small-screen.fi-text--bold {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
}

.c5.fi-text--small-screen.fi-text--lead {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 20px;
  line-height: 1.5;
  font-weight: 400;
}

.c2 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  background-color: hsl(0,0%,100%);
  padding-left: 50px;
  position: relative;
  display: inline-block;
}

.c2 .fi-toggle_icon-container {
  position: absolute;
  margin-right: 10px;
  left: 0;
  top: 0.1em;
}

.c2.fi-toggle--with-button:focus {
  outline: 0;
}

.c2.fi-toggle--with-button:focus .fi-toggle_icon-container:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,55%);
  z-index: 9999;
  border-radius: 14px;
  right: -4px;
  left: -4px;
}

.c2.fi-toggle--with-input:focus-within {
  outline: 0;
}

.c2.fi-toggle--with-input:focus-within .fi-toggle_icon-container:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,55%);
  z-index: 9999;
  border-radius: 14px;
  right: -4px;
  left: -4px;
}

.c2 > .fi-toggle_input {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  position: absolute;
  width: 40px;
  height: 24px;
  opacity: 0;
  z-index: -9999;
  background-color: hsl(0,0%,100%);
}

.c2 .fi-toggle_icon {
  width: 40px;
  height: 24px;
  vertical-align: bottom;
  overflow: visible;
}

.c2 .fi-toggle_icon .icon-toggle_svg__fi-toggle-icon-circle {
  fill: hsl(0,0%,100%);
}

.c2 .fi-toggle_icon .icon-toggle_svg__fi-toggle-icon-knob {
  -webkit-transform: translateX(0%);
  -ms-transform: translateX(0%);
  transform: translateX(0%);
}

.c2 .fi-toggle_icon.fi-toggle_icon--disabled .icon-toggle_svg__fi-toggle-icon-circle {
  fill: hsl(202,7%,97%);
}

.c2 .fi-toggle_icon.fi-toggle_icon--disabled .icon-toggle_svg__fi-toggle-icon-slide {
  fill: hsl(202,7%,93%);
}

.c2 .fi-toggle_icon.fi-toggle_icon--checked .icon-toggle_svg__fi-toggle-icon-knob {
  -webkit-transform: translateX(47%);
  -ms-transform: translateX(47%);
  transform: translateX(47%);
}

.c2 .fi-toggle_icon.fi-toggle_icon--checked .icon-toggle_svg__fi-toggle-icon-slide {
  fill: hsl(166,54%,80%);
}

.c2 .fi-toggle_icon.fi-toggle_icon--checked .icon-toggle_svg__fi-toggle-icon-circle {
  fill: hsl(166,90%,36%);
}

.c2 .fi-toggle_icon.fi-toggle_icon--checked.fi-toggle_icon--disabled .icon-toggle_svg__fi-toggle-icon-circle {
  fill: hsl(166,54%,80%);
}

<button
  aria-disabled="false"
  aria-pressed="false"
  class="c0 c1 fi-toggle--with-button c2 fi-toggle"
  data-testid="toggle"
  id="test-toggle"
  tabindex="0"
  type="button"
>
  <span
    class="c3 fi-toggle_icon-container"
  >
    <svg
      aria-hidden="true"
      class="c4 fi-toggle_icon"
      focusable="false"
      height="1em"
      viewBox="0 0 37 20"
      width="1em"
    >
      <defs>
        <lineargradient
          id="icon-toggle_svg__c"
          x1="50%"
          x2="50%"
          y1="0%"
          y2="99.021%"
        >
          <stop
            offset="0%"
            stop-color="#292929"
            stop-opacity="0"
          />
          <stop
            offset="80%"
            stop-color="#292929"
            stop-opacity="0.02"
          />
          <stop
            offset="100%"
            stop-color="#292929"
            stop-opacity="0.04"
          />
        </lineargradient>
        <lineargradient
          id="icon-toggle_svg__d"
          x1="50%"
          x2="50%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            stop-color="#FFF"
            stop-opacity="0.12"
          />
          <stop
            offset="20%"
            stop-color="#FFF"
            stop-opacity="0.06"
          />
          <stop
            offset="100%"
            stop-color="#FFF"
            stop-opacity="0"
          />
        </lineargradient>
        <filter
          filterUnits="objectBoundingBox"
          height="120%"
          id="icon-toggle_svg__a"
          width="117.5%"
          x="-8.8%"
          y="-8.8%"
        >
          <feoffset
            dy="1"
            in="SourceAlpha"
            result="shadowOffsetOuter1"
          />
          <fegaussianblur
            in="shadowOffsetOuter1"
            result="shadowBlurOuter1"
            stdDeviation="0.25"
          />
          <fecomposite
            in="shadowBlurOuter1"
            in2="SourceAlpha"
            operator="out"
            result="shadowBlurOuter1"
          />
          <fecolormatrix
            in="shadowBlurOuter1"
            result="shadowMatrixOuter1"
            values="0 0 0 0 0.160784314 0 0 0 0 0.160784314 0 0 0 0 0.160784314 0 0 0 0.2 0"
          />
          <feoffset
            in="SourceAlpha"
            result="shadowOffsetOuter2"
          />
          <fegaussianblur
            in="shadowOffsetOuter2"
            result="shadowBlurOuter2"
            stdDeviation="0.5"
          />
          <fecomposite
            in="shadowBlurOuter2"
            in2="SourceAlpha"
            operator="out"
            result="shadowBlurOuter2"
          />
          <fecolormatrix
            in="shadowBlurOuter2"
            result="shadowMatrixOuter2"
            values="0 0 0 0 0.160784314 0 0 0 0 0.160784314 0 0 0 0 0.160784314 0 0 0 0.2 0"
          />
          <femerge>
            <femergenode
              in="shadowMatrixOuter1"
            />
            <femergenode
              in="shadowMatrixOuter2"
            />
          </femerge>
        </filter>
        <circle
          cx="10"
          cy="10"
          id="icon-toggle_svg__b"
          r="10"
        />
      </defs>
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          class="icon-toggle_svg__fi-toggle-icon-slide"
          d="M36.5 10c0 3.9-3.282 7-7.412 7H7.912C3.782 17 .5 13.9.5 10s3.282-7 7.412-7h21.176c4.13 0 7.412 3.1 7.412 7z"
          fill="#A5ADB1"
        />
        <g
          class="icon-toggle_svg__fi-toggle-icon-knob"
        >
          <use
            fill="#000"
            filter="url(#icon-toggle_svg__a)"
            xlink:href="#icon-toggle_svg__b"
          />
          <circle
            class="icon-toggle_svg__fi-toggle-icon-circle"
            cx="10"
            cy="10"
            fill="#FFF"
            r="9.75"
            stroke="url(#icon-toggle_svg__c)"
            stroke-linejoin="square"
            stroke-width="0.5"
          />
          <circle
            cx="10"
            cy="10"
            r="9.75"
            stroke="url(#icon-toggle_svg__d)"
            stroke-linejoin="square"
            stroke-width="0.5"
          />
        </g>
      </g>
    </svg>
  </span>
  <span
    class="c3 fi-text c5 fi-text--body"
  >
    Test
  </span>
</button>
`;

exports[`Toggle variant withInput:  calling render with the same component on the same container does not remount 1`] = `
.c3 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  overflow: visible;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline-block;
  max-width: 100%;
}

.c3::-webkit-input-placeholder {
  color: inherit;
  opacity: 0.54;
}

.c0 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  max-width: 100%;
}

.c4 {
  line-height: 1.15;
  -ms-text-size-adjust: 100%;
  -webkit-text-size-adjust: 100%;
  margin: 0;
  padding: 0;
  border: 0;
  box-sizing: border-box;
  font: 100% inherit;
  line-height: 1;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  vertical-align: baseline;
  color: inherit;
  background: none;
  cursor: inherit;
  display: inline;
  max-width: 100%;
  word-wrap: normal;
  word-break: normal;
  white-space: normal;
}

.c1 {
  cursor: pointer;
}

.c1.fi-toggle--disabled {
  cursor: not-allowed;
}

.c5 {
  display: inline-block;
  vertical-align: baseline;
}

.c6 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  color: hsl(0,0%,16%);
}

.c6.fi-text--bold {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 600;
}

.c6.fi-text--lead {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 22px;
  line-height: 1.5;
  font-weight: 400;
}

.c6.fi-text--small-screen {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 400;
}

.c6.fi-text--small-screen.fi-text--bold {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 16px;
  line-height: 1.5;
  font-weight: 600;
}

.c6.fi-text--small-screen.fi-text--lead {
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 20px;
  line-height: 1.5;
  font-weight: 400;
}

.c2 {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  background-color: hsl(0,0%,100%);
  padding-left: 50px;
  position: relative;
  display: inline-block;
}

.c2 .fi-toggle_icon-container {
  position: absolute;
  margin-right: 10px;
  left: 0;
  top: 0.1em;
}

.c2.fi-toggle--with-button:focus {
  outline: 0;
}

.c2.fi-toggle--with-button:focus .fi-toggle_icon-container:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,55%);
  z-index: 9999;
  border-radius: 14px;
  right: -4px;
  left: -4px;
}

.c2.fi-toggle--with-input:focus-within {
  outline: 0;
}

.c2.fi-toggle--with-input:focus-within .fi-toggle_icon-container:after {
  content: '';
  position: absolute;
  pointer-events: none;
  top: -2px;
  right: -2px;
  bottom: -2px;
  left: -2px;
  border-radius: 2px;
  background-color: transparent;
  border: 0px solid hsl(0,0%,100%);
  box-sizing: border-box;
  box-shadow: 0 0 0 2px hsl(196,77%,55%);
  z-index: 9999;
  border-radius: 14px;
  right: -4px;
  left: -4px;
}

.c2 > .fi-toggle_input {
  color: hsl(0,0%,16%);
  -webkit-letter-spacing: 0;
  -moz-letter-spacing: 0;
  -ms-letter-spacing: 0;
  letter-spacing: 0;
  -webkit-text-decoration: none;
  text-decoration: none;
  word-break: break-word;
  overflow-wrap: break-word;
  -webkit-font-smoothing: antialiased;
  font-family: 'Source Sans Pro','Helvetica Neue','Arial',sans-serif;
  font-size: 18px;
  line-height: 1.5;
  font-weight: 400;
  position: absolute;
  width: 40px;
  height: 24px;
  opacity: 0;
  z-index: -9999;
  background-color: hsl(0,0%,100%);
}

.c2 .fi-toggle_icon {
  width: 40px;
  height: 24px;
  vertical-align: bottom;
  overflow: visible;
}

.c2 .fi-toggle_icon .icon-toggle_svg__fi-toggle-icon-circle {
  fill: hsl(0,0%,100%);
}

.c2 .fi-toggle_icon .icon-toggle_svg__fi-toggle-icon-knob {
  -webkit-transform: translateX(0%);
  -ms-transform: translateX(0%);
  transform: translateX(0%);
}

.c2 .fi-toggle_icon.fi-toggle_icon--disabled .icon-toggle_svg__fi-toggle-icon-circle {
  fill: hsl(202,7%,97%);
}

.c2 .fi-toggle_icon.fi-toggle_icon--disabled .icon-toggle_svg__fi-toggle-icon-slide {
  fill: hsl(202,7%,93%);
}

.c2 .fi-toggle_icon.fi-toggle_icon--checked .icon-toggle_svg__fi-toggle-icon-knob {
  -webkit-transform: translateX(47%);
  -ms-transform: translateX(47%);
  transform: translateX(47%);
}

.c2 .fi-toggle_icon.fi-toggle_icon--checked .icon-toggle_svg__fi-toggle-icon-slide {
  fill: hsl(166,54%,80%);
}

.c2 .fi-toggle_icon.fi-toggle_icon--checked .icon-toggle_svg__fi-toggle-icon-circle {
  fill: hsl(166,90%,36%);
}

.c2 .fi-toggle_icon.fi-toggle_icon--checked.fi-toggle_icon--disabled .icon-toggle_svg__fi-toggle-icon-circle {
  fill: hsl(166,54%,80%);
}

<label
  class="c0 c1 fi-toggle--with-input c2 fi-toggle fi-toggle_label"
  data-testid="toggle"
  for="test-toggle"
>
  <input
    class="c3 fi-toggle_input"
    id="test-toggle"
    type="checkbox"
  />
  <span
    class="c4 fi-toggle_icon-container"
  >
    <svg
      aria-hidden="true"
      class="c5 fi-toggle_icon"
      focusable="false"
      height="1em"
      viewBox="0 0 37 20"
      width="1em"
    >
      <defs>
        <lineargradient
          id="icon-toggle_svg__c"
          x1="50%"
          x2="50%"
          y1="0%"
          y2="99.021%"
        >
          <stop
            offset="0%"
            stop-color="#292929"
            stop-opacity="0"
          />
          <stop
            offset="80%"
            stop-color="#292929"
            stop-opacity="0.02"
          />
          <stop
            offset="100%"
            stop-color="#292929"
            stop-opacity="0.04"
          />
        </lineargradient>
        <lineargradient
          id="icon-toggle_svg__d"
          x1="50%"
          x2="50%"
          y1="0%"
          y2="100%"
        >
          <stop
            offset="0%"
            stop-color="#FFF"
            stop-opacity="0.12"
          />
          <stop
            offset="20%"
            stop-color="#FFF"
            stop-opacity="0.06"
          />
          <stop
            offset="100%"
            stop-color="#FFF"
            stop-opacity="0"
          />
        </lineargradient>
        <filter
          filterUnits="objectBoundingBox"
          height="120%"
          id="icon-toggle_svg__a"
          width="117.5%"
          x="-8.8%"
          y="-8.8%"
        >
          <feoffset
            dy="1"
            in="SourceAlpha"
            result="shadowOffsetOuter1"
          />
          <fegaussianblur
            in="shadowOffsetOuter1"
            result="shadowBlurOuter1"
            stdDeviation="0.25"
          />
          <fecomposite
            in="shadowBlurOuter1"
            in2="SourceAlpha"
            operator="out"
            result="shadowBlurOuter1"
          />
          <fecolormatrix
            in="shadowBlurOuter1"
            result="shadowMatrixOuter1"
            values="0 0 0 0 0.160784314 0 0 0 0 0.160784314 0 0 0 0 0.160784314 0 0 0 0.2 0"
          />
          <feoffset
            in="SourceAlpha"
            result="shadowOffsetOuter2"
          />
          <fegaussianblur
            in="shadowOffsetOuter2"
            result="shadowBlurOuter2"
            stdDeviation="0.5"
          />
          <fecomposite
            in="shadowBlurOuter2"
            in2="SourceAlpha"
            operator="out"
            result="shadowBlurOuter2"
          />
          <fecolormatrix
            in="shadowBlurOuter2"
            result="shadowMatrixOuter2"
            values="0 0 0 0 0.160784314 0 0 0 0 0.160784314 0 0 0 0 0.160784314 0 0 0 0.2 0"
          />
          <femerge>
            <femergenode
              in="shadowMatrixOuter1"
            />
            <femergenode
              in="shadowMatrixOuter2"
            />
          </femerge>
        </filter>
        <circle
          cx="10"
          cy="10"
          id="icon-toggle_svg__b"
          r="10"
        />
      </defs>
      <g
        fill="none"
        fill-rule="evenodd"
      >
        <path
          class="icon-toggle_svg__fi-toggle-icon-slide"
          d="M36.5 10c0 3.9-3.282 7-7.412 7H7.912C3.782 17 .5 13.9.5 10s3.282-7 7.412-7h21.176c4.13 0 7.412 3.1 7.412 7z"
          fill="#A5ADB1"
        />
        <g
          class="icon-toggle_svg__fi-toggle-icon-knob"
        >
          <use
            fill="#000"
            filter="url(#icon-toggle_svg__a)"
            xlink:href="#icon-toggle_svg__b"
          />
          <circle
            class="icon-toggle_svg__fi-toggle-icon-circle"
            cx="10"
            cy="10"
            fill="#FFF"
            r="9.75"
            stroke="url(#icon-toggle_svg__c)"
            stroke-linejoin="square"
            stroke-width="0.5"
          />
          <circle
            cx="10"
            cy="10"
            r="9.75"
            stroke="url(#icon-toggle_svg__d)"
            stroke-linejoin="square"
            stroke-width="0.5"
          />
        </g>
      </g>
    </svg>
  </span>
  <span
    class="c4 fi-text c6 fi-text--body"
  >
    Test
  </span>
</label>
`;
